compat-data-schema.md

```diff
@@ -0,0 +1,374 @@
+third-parties. The new schema is introduced, with some test schemas, third-party
+tools (like MDN macros) are updated to support both the old and new schemas,
+browser compat data is updated to the new schema, and finally the third-party
+support for the old schema is removed.
+
+__Note:__ You cannot mix two schemas in the same file. `"version"` is unique in
+a given file.
+
+### Feature identifier
+A _feature_ is a functionality of the platform. It is an entity that can be used
+independently. A _sub-feature_ is a specific value or behavior of a feature. The
+division is a bit arbitrary, but matches the way developers perceive features of
+a platform
+
+* For CSS, an entity like a property, a pseudo-class, a pseudo-element, an
+at-rule, or a descriptor is a feature. A value, a new syntax change, or a
+specific behavior, like if a property applies to a set of elements or another
+are sub-features.
+
+* For HTML, an element or an attribute are features, while specific values of an
+attribute are sub-features.
+
+* For Web APIs or JavaScript, an interface, a method, a property, a constructor
+are features, while arguments or special values of an enumerated type are
+sub-features.
+
+Each feature is identified by a unique hierarchy of strings. E.g the
+`text-align` property is identified by `css.properties.text-align`.
+
+The strings of an identifier are not intended to be displayed and are therefore
+not translatable.
+
+In JSON, this gives:
+
+    {
+      "css": {
+        "properties": {
+          "text-align": {…},
+          …
+        },
+        …
+      },
+      …
+    }
+
+ That way, each feature is uniquely identifier, independently of the file it is
+ defined in.
+
+ The hierarchy of identifiers is not defined inside the schema. It is a
+ convention of the project using the schema.
+
+ In the MDN browser-compat case, it is:
+
+ _Note: this list will evolve as we migrate our data_
+
+     {
+       "css": {
+         "properties": {…}
+         "pseudo-classes": {…}
+         "pseudo-elements": {…}
+         "at-rules": {…}
+       },
+       "html": {
+         "elements": {
+           …,
+           "<element-name>": {
+             …,
+             "<attribute-name>": {…},
+             …
+           }
+         }
+         "global_attributes": {…}  
+       }
+     }
+
+### Feature description
+
+A feature is represented by an identifier containing the `"__compat"` property.
+In this compat property, you'll find the list of sub-features associated to the
+feature.
+
+A feature has at least one sub-feature, representing the basic support. It is
+always named `"basic"`.
+
+The basic support feature represents the minimal set of functionality included
+when a feature is qualified of 'supported'. What this represents depends of the
+evolution of the feature over time, both in term of specification and of browser
+support. Another way of seeing it is to consider `"support"` as representing all
```

Comment:
Actually the schema calls it `"basic_support"` I think? https://github.com/teoli2003/browser-compat-data/blob/schema-with-identifier/compat-data.schema.json#L100